src/maybe.ts

```diff
@@ -1132,6 +1156,20 @@ export function toString<T>(maybe: Maybe<T>): string {
   return `${maybe.variant}${body}`;
 }
 
+/**
+ * Create an `Object` representation of a `Maybe` instance.
+ * 
+ * Useful for serialization. `JSON.stringify()` uses it.
+ * 
+ * @param maybe The value to convert to JSON
+ * @returns     The JSON representation of the `Maybe`
+ */
+export function toJSON(maybe: Maybe<any>): MaybeJsonShape {
```

Comment:
This function will need two changes: `MaybeJSON<T>` as the return type, and simply to do `value: maybe.value` in the `Just` branch.

```suggestion
export function toJSON<T>(maybe: Maybe<T>): MaybeJSON<T> {
```